### PR TITLE
Fix duplicate r_core_get_boundaries_prot lookups in cmd_search WIP

### LINKED ISSUE #11097: duplicate r_core_get_boundaries_prot lookups in cmd_search
in cmd.search.c cmd_search does r_core_get_boundaries_prot to create a list of boundaries at the top level before the switch statement. 
https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L2598

Then each of these functions do their own (duplicate) r_core_get_boundaries_prot lookup
* [/V](https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L3012)
* [search_hash](https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L3245)
* [r_core_search_rop](https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L2704-L2715)
* [do_asm_search](https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L3368-L3370)
* [search_similar_pattern](https://github.com/radare/radare2/blob/master/libr/core/cmd_search.c#L2983)

Im not very familiar with the code, but assuming any of them dont need to generate a different set of boundaries then perhaps they could be refactored to be passed/use the existing boundaries?

Or we can NOT do the lookup at the top so theres no chance of duplication, and expect each command to do its own...

### PR DESCRIPTION
**Pull request by jacobrosenthal**
breaking everything im sure, but it does build.. really though, dont merge.

Closes #11097

### CODE REVIEW

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -170,17 +170,12 @@ static void cmd_search_init(RCore *core) {
 	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, /x, slash_x);
 }
 
-static int search_hash(RCore *core, const char *hashname, const char *hashstr, ut32 minlen, ut32 maxlen) {
+static int search_hash(RCore *core, const char *hashname, const char *hashstr, ut32 minlen, ut32 maxlen, struct search_parameters *param) {
```
Comment (radare) [inline libr/core/cmd_search.c]: why not use the short camelcase name?

**Comment by radare** [comment]
doesnt seems to break any test and the change sounds reasonable to me

**Comment by radare** [comment]
what's left? (for the WIP)

**Comment by radare** [comment]
btw, commits start in uppercase

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -1319,34 +1310,12 @@ static int r_core_search_rop(RCore *core, RInterval search_itv, int opt, const c
...
-			result = false;
-			goto bad;
-		}
-		map->fd = r_io_fd_get_current (core->io);
-		map->itv.addr = from;
-		map->itv.size = to - from;
-		list = r_list_newf (free);
-		r_list_append (list, map);
-	}
-
```
Comment (jacobrosenthal) [inline libr/core/cmd_search.c]: Basically I worry that I dont understand this giant block here and below to know if Ive replaced its functionality properly. If it looks good to you all then fine by me.

File: libr/core/cmd_search.c (hunk 1 above)
Comment (jacobrosenthal) [reply libr/core/cmd_search.c]: That decision predates me. This is a previously existing struct.

**Comment by jacobrosenthal** [comment]
Added do_syscall_search fix and changed commit name

**Comment by radare** [comment]
please rebase and solve the conflicts

File: libr/core/cmd_search.c (hunk 1 above)
Comment (radare) [reply libr/core/cmd_search.c]: oh i see there's no cameltoe case for this struct

**Comment by radare** [comment]
i'll merge after you fix the conflicts. thanks!

**Comment by radare** [comment]
i solved the conflicts

File: libr/core/cmd_search.c [hunk 3]
```diff
@@ -2079,6 +2056,9 @@ static void do_asm_search(RCore *core, struct search_parameters *param, const ch
 		input ++;
 	}
 	r_list_foreach (param->boundaries, itermap, map) {
+		if (!r_itv_overlap (search_itv, map->itv)) {
```
Comment (radare) [inline libr/core/cmd_search.c]: not sure if we want to skip the whole interval just because it overlaps a portion of another map..

**Comment by radare** [comment]
Also. Any DOS virus would work too. As long as real mode x86 prgrams interact with bios interrupts directly

File: libr/core/cmd_search.c [hunk 4]
```diff
@@ -2055,22 +2045,9 @@ static void do_asm_search(RCore *core, struct search_parameters *param, const ch
...
-	param->boundaries = r_core_get_boundaries_prot (core, -1, param->mode, "search");
 	maxhits = (int) r_config_get_i (core->config, "search.maxhits");
 	filter = (int) r_config_get_i (core->config, "asm.filter");
 
-	if (!param->boundaries) {
-		map = R_NEW0 (RIOMap);
-		if (map) {
-			map->fd = r_io_fd_get_current (core->io);
-			map->itv.addr = r_config_get_i (core->config, "search.from");
-			map->itv.size = r_config_get_i (core->config, "search.to") - map->itv.addr;
```
Comment (radare) [inline libr/core/cmd_search.c]: Is search.from/to honored in params already?

Comment (jacobrosenthal) [reply libr/core/cmd_search.c]: My thinking was its captured by search_itv

**Comment by radare** [comment]
btw, as a side note, if a PR fixes an issue put the commit like this: "Fix #issuenumber -description" this way github will autoclose the issue when merged

**Pull request merged**